### ISSUE TITLE
Annotations: Annotation buttons don't have accessible names

### ISSUE THREAD

**Issue by fastfrwrd** [grafana/grafana#119650]
**Steps to reproduce**: Open a dashboard with annotations. Look at the buttons for annotations (they are the triangular buttons along the x axis in this screenshot)

<img width="752" height="330" alt="Image" src="https://github.com/user-attachments/assets/3893ad5a-3e07-448f-abc1-7d1dfad86112" />

**Actual Result**: The annotation buttons should have some kind of name for screenreaders

**Expected Result**: Annotation buttons are missing names. 

**Relevant WCAG Criteria:** https://dequeuniversity.com/rules/axe/4.11/button-name

**Environment**:
- Grafana version: 12.4.0
- Data source type & version: used a Timeseries
- User OS & Browser: Chrome
- Others: n/a
